### PR TITLE
v3.0.0

### PR DESCRIPTION
**Pull request by alphachai**
**tl;dr**: Improved reliability, readability, and coverage. Breaking changes are largely superficial. Removed Mintel-specific modules.

## Release Notes
- Reduced code complexity
- Added type hints
- Increased test coverage
- Bugfix: lpipe.sqs.delete_message_batch called boto3 incorrectly
- Breaking: sentry-sdk no longer a required dependency, install with `lpipe = {"extras": ["sentry"], "version": "*"}`, and enable it by setting `process_event(exception_handler=lpipe.contrib.sentry.capture)`
- Breaking: moved `lpipe.sentry` to `lpipe.contrib.sentry`
- Breaking: moved `lpipe._boto3` to `lpipe.contrib.boto3`
- Breaking: removed `lpipe.testing`, use the boto3-fixtures library instead
- Breaking: removed `lpipe.taxonomy`
- Breaking: renamed LpipeBaseException to LPBaseException (although, you shouldn't use the base class)
- Breaking: renamed ServerlessLogger to LPLogger
- Breaking: renamed `lpipe.utils._repr` to `repr`
- Semi-Breaking: moved Action, Payload, and Queue classes into their own modules, but they're still imported into the top level module (i.e. `lpipe.Action` still works)
- Deprecated: moved `lpipe.utils` get_nested, set_nested to `lpipe.contrib.mindictive`, stubs still in place

### CODE REVIEW

File: dummy_lambda/func/main.py [hunk 1]
```diff
@@ -53,23 +53,26 @@ def return_foobar(**kwargs):
...
 def throw_exception(**kwargs):
-    try:
-        raise Exception("Test event. Please ignore.")
-    except Exception as e:
-        sentry.capture(e)
-        raise FailButContinue from e
+    raise Exception()
+
+
+#     try:
```
Comment (chavlin) [inline dummy_lambda/func/main.py:71]: nit: should remove commented-out code from production-ready code.

Comment (alphachai) [reply dummy_lambda/func/main.py:71]: This is test code and [is excluded](https://github.com/mintel/lpipe/blob/v3.0.0/setup.py#L39) from the built package, but I do agree with the sentiment. In this specific case, I'm going to open another PR to re-activate these tests.
